### DISCUSSION TITLE
check permission model id specific.

### DISCUSSION THREAD

**Discussion by technoknol** [spatie/laravel-permission#2189] [Q&A]
How can I check permission of user model specific, like if `$comment->author == $user->id` then it should allow a permission (e.g. `edit-own-comment`) ?

**Comment by erikn69** [comment]
You could register your own gate logic with a service provider
https://github.com/spatie/laravel-permission/blob/67e54e2c7941d7aaa93c207549cdb7c5a44c68f5/src/PermissionServiceProvider.php#L26-L29
Or maybe using after
https://spatie.be/docs/laravel-permission/v5/basic-usage/super-admin#content-gateafter

**Comment by technoknol** [comment]
Thanks for quick answer. What if I have dynamic permissions and not only one?

**Reply by angeljqv** [reply]
use conditionals
